cooked-validators/src/Cooked/Tx/Constraints.hs

```diff
@@ -169,7 +171,15 @@ outConstraintToTxOut (PaysScript validator msc datum value) =
 orderTxOutputs :: [OutConstraint] -> [Pl.TxOut] -> [Pl.TxOut]
 orderTxOutputs expected given =
   let res = map outConstraintToTxOut expected
-   in res ++ (given List.\\ res)
+   in -- Need to have a custom equality, filtering out anything that pays to a script
+      -- as well, even if Nothing is the sc.
+      -- That is, PaysScript val Nothing datum value == PaysScript val (Just _) datum value
+      -- NB: this is a huge hack, should *NOT* be merged to main. We must just use actual
+      -- equality, instead of this filtering with eraseStakingCredential
+      res ++ (given List.\\ map eraseStakingCredential res)
+  where
+    eraseStakingCredential txOut =
+      txOut {Pl.txOutAddress = (Pl.txOutAddress txOut) {Pl.addressStakingCredential = Nothing}}
 
```

Comment:
As the comment suggestions, this should not be merged to main as is. However, I'm not sure what the "correct" fix is. We could modify the equality instance of PaysScript to ignore StakingCredential there, but that seems like it could lead to unexpected behavior later on. Anyone have any ideas?

Reply:
I'm not fully sure, but if you're basically looking at plugging custom equality into `\\`, you could also use [`List.deleteFirstsBy`](https://hackage.haskell.org/package/base-4.16.3.0/docs/Data-List.html#v:deleteFirstsBy).

Reply:
I'm not entirely sure either what exactly we meant when we wrote this comment. Either way, `List.deleteFirstsBy` is a much cleaner implementation, so I changed it to that for now at least